src/algebra/char_p/localization.lean

```diff
@@ -0,0 +1,67 @@
+/-
+Copyright (c) 2021 Jon Eugster. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jon Eugster
+-/
+import algebra.char_p.basic
+import ring_theory.localization
+
+
+/-!
+# Characteristics of field of fractions
+
+In this file we describe the characteristics of the field of fractions of an integral domain.
+
+
+## Main results
+
+If `R` is an `integral_domain` then the `fraction_ring R` has the same characteristic as `R`.
+
+
+## TODO
+
+* Could generalise this to any `R`-algebra over a `comm_semiring R`
+  with injective algebra map.
```

Comment:
I think you should do this generalization,
```
def char_p_of_injective_algebra_map {R A} [comm_ring R] [semiring A] [algebra R A]
  (h : function.injective (algebra_map R A)) (p : ℕ) [char_p R p] : char_p A p :=
{ cast_eq_zero_iff := λ x,
  begin
    rw ←char_p.cast_eq_zero_iff R p x,
    change algebra_map ℕ A x = 0 ↔ algebra_map ℕ R x = 0,
    rw is_scalar_tower.algebra_map_apply ℕ R A x,
    split,
    { apply (ring_hom.injective_iff _).mp h _ },
    { rintro h, rw [h, ring_hom.map_zero], }
  end }
```

Reply:
Since then the result you want is just:
```
instance char_p [char_p R p] : char_p (fraction_ring R) p :=
char_p_of_injective_algebra_map (fraction_map.injective (fraction_ring.of R)) p
```

Reply:
Thank you, I'll incorporate that.